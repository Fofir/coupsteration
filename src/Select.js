import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Select extends Component {
  onChange = (evt) => {
    this.props.onChange(evt.target.value);
  }

  render() {
    const { disabled, options } = this.props;
    return (
      <select
        onChange={this.onChange}
        disabled={disabled}
      >
        {options.map(({ value, label }) => <option key={value} value={value}>{label}</option>)}
      </select>
    );
  }
};

Select.propTypes = {
  onChange: PropTypes.func.isRequired,
  options: PropTypes.arrayOf(PropTypes.shape({
    label: PropTypes.string.isRequired,
    value: PropTypes.oneOfType([PropTypes.number, PropTypes.string]).isRequired,
  })).isRequired,
  disabled: PropTypes.bool.isRequired,
};

Select.defaultProps = {
  options: [],
  disabled: false,
};

export default Select;
