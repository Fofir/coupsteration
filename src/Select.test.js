import React from 'react';
import Select from './Select';
import renderer from 'react-test-renderer';
import { configure, shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });

describe('<Select />', () => {
  it('renders correctly with options', () => {
    const options = [{
      label: 'Type 1',
      value: 'type 1',
    }, {
      label: 'Type 2',
      value: 'type 2',
    }];

    const tree = renderer
      .create(<Select options={options} onChange={jest.fn()} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly when disabled', () => {
    const options = [{
      label: 'Type 1',
      value: 'type 1',
    }, {
      label: 'Type 2',
      value: 'type 2',
    }];

    const tree = renderer
      .create(<Select options={options} disabled onChange={jest.fn()} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with a custom className', () => {
    const options = [{
      label: 'Type 1',
      value: 'type 1',
    }, {
      label: 'Type 2',
      value: 'type 2',
    }];

    const tree = renderer
      .create(<Select options={options} className="my-select-class" disabled onChange={jest.fn()} />)
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('shold call props.onChange with the new select value upon change', () => {
    const options = [{
      label: 'Type 1',
      value: 'type 1',
    }, {
      label: 'Type 2',
      value: 'type 2',
    }];
    const onChange = jest.fn();
    const wrapper = shallow(<Select options={options} onChange={onChange} />);
    const instance = wrapper.instance();
    instance.forceUpdate(); // to make sure the bound class method is spied on
    wrapper.find('select').simulate('change', { target: { value: 'type 2' }});
    expect(onChange).toHaveBeenCalledWith('type 2');
  });
});
