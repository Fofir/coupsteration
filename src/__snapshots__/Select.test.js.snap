// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Select /> renders correctly when disabled 1`] = `
<select
  disabled={true}
  onChange={[Function]}
>
  <option
    value="type 1"
  >
    Type 1
  </option>
  <option
    value="type 2"
  >
    Type 2
  </option>
</select>
`;

exports[`<Select /> renders correctly with options 1`] = `
<select
  disabled={false}
  onChange={[Function]}
>
  <option
    value="type 1"
  >
    Type 1
  </option>
  <option
    value="type 2"
  >
    Type 2
  </option>
</select>
`;
